cdap-ui/app/cdap/components/AbstractWidget/DLPCustomWidget/DLPRow.tsx

```diff
@@ -0,0 +1,398 @@
+            }
+          : {};
+      this.setState({ ...this.state, ...jsonObj });
+    } catch (error) {
+      this.setState({
+        fields: '',
+        transform: '',
+        filters: '',
+        transformProperties: {},
+      });
+    }
+  }
+
+  // Handles onChange for MultiSelect components (filters and fields selectors)
+  private handleChangeMultiSelect = (type: StateKeys, e: string) => {
+    if (type === 'filters') {
+      const noneWasSelected: boolean = this.state.filters.includes('NONE');
+      const noneIsSelected: boolean = e.includes('NONE');
+
+      // Ensuring user can't select 'NONE' along with other filters
+      // TODO: Use another component for the filters selector (https://issues.cask.co/browse/CDAP-16124)
+      if (noneWasSelected && noneIsSelected) {
+        e = e.replace('NONE,', '');
+      } else if (!noneWasSelected && noneIsSelected) {
+        e = 'NONE';
+      }
+    }
+
+    this.handleChange({ [type]: e });
+  };
+
+  // Handles onChange for Select components (transform selector)
+  private handleChangeTransformSelect = (type: StateKeys, e) => {
+    const value = e.target.value;
+    if (value === this.state.transform) {
+      return;
+    }
+    const newValues = { [type]: value };
+
+    const transformProps = {};
+    this.props.transforms
+      .filter((transform) => transform.name === value)[0]
+      .options.forEach((element) => {
+        transformProps[element.name] = element['widget-attributes'].default || '';
+      });
+    newValues.transformProperties = transformProps;
+    newValues.expanded = Object.keys(transformProps).length > 0;
+    newValues.transformObj = this.props.transforms.filter((t) => t.name === value)[0] || null;
+    this.configurationGroupPluginProperties = parseTransformOptions(newValues.transformObj);
+    this.configurationGroupWidgetJson =
+      newValues.transformObj !== null
+        ? {
+            'configuration-groups': [
+              {
+                label: newValues.transformObj.label + ' properties',
+                properties: newValues.transformObj.options,
+              },
+            ],
+            filters: newValues.transformObj.filters, // Passing ConfigurationGroup filters to allow for dynamic showing/hiding of properties
+          }
+        : {};
+    this.handleChange(newValues);
+  };
+
+  private handleChangeTransformOptions(values: Record<string, string>) {
+    // Set values in state if they were cleared by the configuration group
+    if (this.state.transformObj) {
+      this.state.transformObj.options.forEach((option: IWidgetProperty) => {
+        if (values[option.name] === undefined) {
+          values[option.name] = option['widget-attributes'].default || '';
+        }
+      });
+    }
+
+    this.handleChange({ transformProperties: values });
+  }
+
+  private handleChange = (newValues) => {
+    this.setState(
+      {
+        ...this.state,
```

Comment:
this can be just ```this.setState(newValues)```. React will be smart to update only those properties that we set and leave the rest of the properties unchanged.

Reply:
oh perfect, done